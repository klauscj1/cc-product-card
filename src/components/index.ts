import { ProductCard as ProductCardHOC } from "./ProductoCard";
import { ProductTitle } from "./ProductTitle";
import { ProductImage } from "./ProductoImage";
import { ProductButtons } from "./ProductButtons";
import { ProductCardHOCProps } from "../interfaces/interfaces";
export { ProductButtons } from "./ProductButtons";
export { ProductImage } from "./ProductoImage";
export { ProductTitle } from "./ProductTitle";

export const ProductCard: ProductCardHOCProps = Object.assign(ProductCardHOC, {
  Title: ProductTitle,
  Image: ProductImage,
  Buttons: ProductButtons,
});

export default ProductCard;
