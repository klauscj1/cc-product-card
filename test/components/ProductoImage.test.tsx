import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components/ProductoCard';
import { ProductImage } from '../../src/components/ProductoImage';
import { product2 } from '../data/product';

describe('ProductImage', () => {
  test('debe mostrar el componente correctamente con el titulo personalizado', () => {
    const wrapper = renderer.create(
      <ProductImage img="./assets/no-image.png" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe mostrar el nombre del producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
