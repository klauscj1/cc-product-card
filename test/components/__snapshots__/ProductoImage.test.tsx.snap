// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe mostrar el componente correctamente con el titulo personalizado 1`] = `
<img
  alt="product-image"
  className="productImg undefined"
  src="./assets/no-image.png"
/>
`;

exports[`ProductImage debe mostrar el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="product-image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImate debe mostrar el componente correctamente con el titulo personalizado 1`] = `
<img
  alt="product-image"
  className="productImg undefined"
  src="./assets/no-image.png"
/>
`;

exports[`ProductImate debe mostrar el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="product-image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductTitle debe mostrar el componente correctamente con el titulo personalizado 1`] = `
<img
  alt="product-image"
  className="productImg undefined"
  src="./assets/no-image.png"
/>
`;

exports[`ProductTitle debe mostrar el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="product-image"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
